cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -104,28 +125,33 @@ func mutateEtcdOrDie(mutator func() error) {
 	}
 }
 
-func newEtcdClient() (client *etcd.Client) {
-	maxConnectRetries := 12
-	for maxConnectRetries > 0 {
-		if _, err := tools.GetEtcdVersion(*etcd_server); err != nil {
-			log.Fatalf("Failed to connect to etcd server: %v, error: %v", *etcd_server, err)
-			if maxConnectRetries > 0 {
-				log.Println("Retrying request after 5 second sleep.")
-				time.Sleep(5 * time.Second)
-				maxConnectRetries--
-			} else {
-				return nil
-			}
-		} else {
-			log.Printf("Etcd server found: %v", *etcd_server)
+func newEtcdClient(etcdServer string) (*etcd.Client, error) {
+	var (
+		client *etcd.Client
+		err    error
+	)
+	retries := maxConnectRetries
+	for retries > 0 {
+		if _, err = tools.GetEtcdVersion(etcdServer); err == nil {
+			break
+		}
+		if maxConnectRetries == 1 {
```

Comment:
I am fixing it in a follow up PR with some real tests. Good catch!

On Tue, May 12, 2015 at 11:03 AM, Daniel Smith <notifications@github.com>
wrote:

> In cluster/addons/dns/kube2sky/kube2sky.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7901#discussion_r30162136>
> :
>
> > -			} else {
> > -				return nil
> > -			}
> > -		} else {
> > -			log.Printf("Etcd server found: %v", *etcd_server)
> > +func newEtcdClient(etcdServer string) (*etcd.Client, error) {
> > +	var (
> > +		client *etcd.Client
> > +		err    error
> > +	)
> > +	retries := maxConnectRetries
> > +	for retries > 0 {
> > +		if _, err = tools.GetEtcdVersion(etcdServer); err == nil {
> > +			break
> > +		}
> > +		if maxConnectRetries == 1 {
>
> Not the end of the world... If maxConnectRetries == 1, it'll just never
> print out the log message, and when it runs out of retries, it'll wait an
> extra 5 seconds. @vishh <https://github.com/vishh> Worth fixing in a
> followup, though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7901/files#r30162136>
> .
>